README.md

```diff
@@ -1,95 +1,179 @@
-   </td>
-   <td><strong>INT4 (AWQ/GPTQ/bnb)</strong>
-   </td>
-  </tr>
-  <tr>
-   <td>8B
-   </td>
-   <td>16 GB
-   </td>
-   <td>8 GB
-   </td>
-   <td>4 GB
-   </td>
-  </tr>
-  <tr>
-   <td>70B
-   </td>
-   <td>140 GB
-   </td>
-   <td>70 GB
-   </td>
-   <td>35 GB
-   </td>
-  </tr>
-  <tr>
-   <td>405B
-   </td>
-   <td>810 GB
-   </td>
-   <td>405 GB
-   </td>
-   <td>203 GB
-   </td>
-  </tr>
-</table>
-
-_Note: These are estimated values and may vary based on specific implementation details and optimizations._
-
-Here are some notebooks to help you started:
-
-* Run Llama 8B in free Google Colab in half precision
-* [Run Llama 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
-* [Run Llama 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
-* [Run Llama 8B with AWQ & fused ops](./awq.ipynb)
+Would you like to run inference of the Llama models locally?
+So do we! The memory requirements depend on the model size and the
+precision of the weights. Here's a table showing the approximate
+memory needed for different configurations:
+
+### Llama 3.1
+
+| Model Size | Llama Variant | BF16/FP16 | FP8 | INT4(AWQ/GPTQ/bnb) |
+| :--: | :--: | :--: | :--: | :--: |
+| 1B | 3.2 | 2.5 GB | 1.25GB | 0.75GB |
+| 3B | 3.2 |6.5 GB | 3.2GB | 1.75GB |
+| 8B | 3.1 |16 GB | 8GB | 4GB |
+| 70B | 3.1 | 140 GB | 70GB | 35GB |
+|405B | 3.1 |810 GB | 405GB | 204GB |
+
+
+> [!NOTE]
+> These are estimated values and may vary based on specific
+implementation details and optimizations.
+
+Working with the capable Llama 3.1 8B models:
+
+* [Run Llama 3.1 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
+* [Run Llama 3.1 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
+* [Run Llama 3.1 8B with AWQ & fused ops](./awq.ipynb)
+
+Working on the 🐘 big Llama 3.1 405B model:
+
 * [Run Llama 3.1 405B FP8](./fp8-405B.ipynb)
 * [Run Llama 3.1 405B quantized to INT4 with AWQ](./awq_generation.py)
 * [Run Llama 3.1 405B quantized to INT4 with GPTQ](./gptq_generation.py)
-* [Run assisted decoding with Llama 405B and Llama 8B](./assisted_decoding.py)
+
+## Model Fine Tuning:
+
+It is often not enough to run inference on the model. 
+Many times, you need to fine-tune the model on some 
+custom dataset. Here are some scripts showing 
+how to fine-tune the models.
+
+Fine tune models on your custom dataset:
+* [Fine tune Llama 3.2 Vision on a custom dataset](./Llama-Vision%20FT.ipynb)
+* [Supervised Fine Tuning on Llama 3.2 Vision with TRL](./sft_vlm.py)
+* [How to fine-tune Llama 3.1 8B on consumer GPU with PEFT and QLoRA with bitsandbytes](./peft_finetuning.py)
+* [Execute a distributed fine tuning job for the Llama 3.1 405B model on a SLURM-managed computing cluster](./qlora_405B.slurm)
+
+## Assisted Decoding Techniques
+
+Do you want to use the smaller Llama 3.2 models to speedup text generation
+of bigger models? These notebooks showcase assisted decoding (speculative decoding), which gives you ~2x speedups for text generation on Llama 3.1 70B (with greedy decoding).
```

Comment:
```suggestion
of bigger models? These notebooks showcase assisted decoding (speculative decoding), which gives you upto 2x speedups for text generation on Llama 3.1 70B (with greedy decoding).
```

nit: Assistive decoding is very sensitive to the nature, so if there's a topic that's out of distribution for draft model, it'd infact be slower than straight up `bf16`

Reply:
Do you think I should add this nit?

Reply:
It's a minor nit, up-to you.

Reply:
I think I will add this nit in the assisted decoding notebook -- where people might be more familiar with why the nit was added in the first place.